lib/modules/manager/pip-compile/common.ts

```diff
@@ -217,3 +220,57 @@ function throwForUnknownOption(arg: string): void {
   }
   throw new Error(`Option ${arg} not supported (yet)`);
 }
+
+function buildRegistryUrl(url: string): URL {
+  const hostRule = hostRules.find({ url });
+  const ret = new URL(url);
+  if (!ret.username) {
+    ret.username = hostRule.username ?? '';
+    ret.password = hostRule.password ?? '';
+  }
+  return ret;
+}
+
+function getRegistryUrlVarFromUrls(
+  varName: keyof GetRegistryUrlVarsResult['environmentVars'],
+  urls: URL[],
+): GetRegistryUrlVarsResult {
+  let haveCredentials = false;
+  for (const url of urls) {
+    if (url.username) {
```

Comment:
```suggestion
    if (url.username || url.password) {
```

can it only have a password?